src/hooks/useSearchParamState.ts

```diff
@@ -105,9 +106,40 @@ const getAllCurrentParams = (searchParams: URLSearchParams) => {
   return allUrlParams;
 };
 
-const useSearchParamsState = (paramsDefinition: SearchParamsStateType) => {
+/**
+ * Works like useState but stores the state in URL search params. Adapted
+ * from https://github.com/jschwindt/react-use-search-params-state/tree/main
+ *
+ * @param paramsDefinition maps search param names to their definitions,
+ * indicating their type and default value
+ * @param initial initial values to set in the search params
+ */
+const useSearchParamsState = (
+  paramsDefinition: SearchParamsStateType,
+  initial?: URLSearchParamsInit
+) => {
```

Comment:
Suggest updating this hook to accept a typed object, so it's easier to understand what the params are from the calling components